### PR TITLE
feat: saved queries with execution info

### PR DESCRIPTION
**Pull request by dpgaspar**
### SUMMARY
Propagates the number of retrieved rows and timestamps to a "related" saved query when a SQLLab query executes.
This is a nice to have UI/UX information for a future home screen on Superset 

### ADDITIONAL INFORMATION
- [ ] Has associated issue:
- [ ] Changes UI
- [ ] Requires DB Migration.
- [X] Confirm DB Migration upgrade and downgrade tested.
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

**Comment by mistercrunch** [comment]
Did we consider having this as a foreign key to the query table? There are probably pros/cons here.

File: tests/sqllab_tests.py [hunk 1]
```diff
@@ -71,6 +71,36 @@ def test_sql_json(self):
...
+                .filter(
+                    SavedQuery.db_id == examples_db_id, SavedQuery.sql == sql_statement
+                )
+                .one_or_none()
+            )
+            assert saved_query_.rows is not None
+            assert saved_query_.last_run == datetime.now()
+
+            # Rollback changes
+            db.session.delete(saved_query_)
```
Comment (villebro) [inline tests/sqllab_tests.py:101]: Nit: I wonder if it were cleaner to just do `db.session.add(SavedQuery(db_id=examples_db_id, sql=sql_statement))` in the beginning, and later name the the queried `SavedQuery` without the trailing underscore, as the original `saved_query` isn't really reused.

**Review by villebro — APPROVED** [review]
LGTM with minor naming comment.

File: tests/sqllab_tests.py (hunk 1 above)
Comment (dpgaspar) [reply tests/sqllab_tests.py:101]: Would look better, yes

**Comment by dpgaspar** [comment]
> Did we consider having this as a foreign key to the query table? There are probably pros/cons here.

@mistercrunch 

I have considered using a custom SQLAlchemy relation on SavedQuery, but I think is not clean since there is no relation between savedqueries and queries, this is kind of an best effort. But more importantly using a relation on savedquery to join query is expensive (since query table grows a lot and we are joining by `sql` text field). 
I think that this write approach is light and not too "evasive"

Thoughts?

**Review by willbarrett — APPROVED** [review]

**Pull request merged**